types/mongoose-delete/mongoose-delete-tests.ts

```diff
@@ -1,49 +1,49 @@
 import mongoose = require('mongoose');
-import mongoose_delete = require('mongoose-delete');
+import MongooseDelete = require('mongoose-delete');
 
-interface PetDocument extends mongoose_delete.SoftDeleteDocument {
+interface PetDocument extends MongooseDelete.SoftDeleteDocument {
   name: string;
 }
 const PetSchema = new mongoose.Schema<PetDocument>({
   name: String,
 });
 // Override all methods
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all' });
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all' });
 // or
-PetSchema.plugin(mongoose_delete, { overrideMethods: true });
+PetSchema.plugin(MongooseDelete, { overrideMethods: true });
 
 // Overide only specific methods
-PetSchema.plugin(mongoose_delete, {
+PetSchema.plugin(MongooseDelete, {
   overrideMethods: ['count', 'find', 'findOne', 'findOneAndUpdate', 'update'],
 });
 // or
-PetSchema.plugin(mongoose_delete, {
+PetSchema.plugin(MongooseDelete, {
   overrideMethods: ['count', 'countDocuments', 'find'],
 });
 // or (unrecognized method names will be ignored)
-PetSchema.plugin(mongoose_delete, { overrideMethods: ['count', 'find', 'errorXyz'] }); // $ExpectError
+PetSchema.plugin(MongooseDelete, { overrideMethods: ['count', 'find', 'errorXyz'] });
 
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedAt: true });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true });
-PetSchema.plugin(mongoose_delete, {
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedAt: true });
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true });
+PetSchema.plugin(MongooseDelete, {
   overrideMethods: 'all',
   deletedByType: String,
 });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true, indexFields: ['deleted'] });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true, indexFields: 'all' });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true, indexFields: 'invalid' }); // $ExpectError
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true, indexFields: ['deleted'] });
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true, indexFields: 'all' });
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true, indexFields: 'invalid' });
```

Comment:
Why was the `// $ExpectError` removed? Shouldn't this call still issue an error?

Reply:
in `@types/mongoose@5.10`, `plugin` function has an overload `plugin<T>(plugin: (schema: Schema, options: T) => void, opts: T): this;)`. So the 'opts' params can be limit to `boolean | 'all' | MongooseDelete.overridableMethods[]`. But in `mongoose`, It's `plugin(fn: (schema: Schema<DocType>, opts?: any) => void, opts?: any): this;`, is an `any`. So should not issue an error.

Reply:
Ick - that's a pretty bad UX they've regressed to, but OK.